Wrappers/Python/cil/framework/labels.py

```diff
@@ -0,0 +1,434 @@
+        ----------
+        engine : str
+            The engine to get the default dimension order for.
+
+        Returns
+        -------
+        list
+            The default dimension order for the given engine.
+
+        Raises
+        ------
+        ValueError
+            If the engine is not a valid member of the Backends enumeration
+        """
+
+        engine_orders = {
+            Backends.ASTRA.value: [cls.CHANNEL.value, cls.VERTICAL.value, \
+                                   cls.ANGLE.value, cls.HORIZONTAL.value],
+            Backends.TIGRE.value: [cls.CHANNEL.value, cls.ANGLE.value, \
+                                   cls.VERTICAL.value, cls.HORIZONTAL.value],
+            Backends.CIL.value: [cls.CHANNEL.value, cls.ANGLE.value, \
+                                 cls.VERTICAL.value, cls.HORIZONTAL.value]
+        }
+        Backends.validate(engine)
+        engine = Backends.get_enum_value(engine)
+
+        return engine_orders[engine]
+
+    @classmethod
+    def get_order_for_engine(cls, engine, geometry):
+        """
+        Returns the order of dimensions for a specific engine and geometry.
+
+        Parameters:
+        ----------
+        engine : str
+            The engine name.
+        geometry : AcquisitionGeometry
+
+        Returns:
+        --------
+        list
+            The order of dimensions for the given engine and geometry.
+        """
+
+        dim_order = cls.get_default_order_for_engine(engine)
+        dimensions = [label for label in dim_order if label in geometry.dimension_labels ]
+
+        return dimensions
+
+    @classmethod
+    def check_order_for_engine(cls, engine, geometry):
+        """
+        Checks if the order of dimensions is correct for a specific engine and geometry.
+
+        Parameters:
+        ----------
+        engine : str
+            The engine name.
+        geometry : AcquisitionGeometry
+
+        Returns:
+        --------
+        bool
+            True if the order of dimensions is correct, False otherwise.
+
+        Raises:
+        -------
+        ValueError
+            If the order of dimensions is incorrect.
+        """
+
+        order_requested = cls.get_order_for_engine(engine, geometry)
+
+        if order_requested == list(geometry.dimension_labels):
+            return True
+        else:
+            raise ValueError(
+                f"Expected dimension_label order {order_requested}, \
+                got {list(geometry.dimension_labels)}.\n\
+                Try using `data.reorder('{engine}')` to permute for {engine}")
+
+class FillTypes(_LabelsBase):
+    """
+    Available fill types for image data.
+
+    Attributes
+    ----------
+    RANDOM ('random'): Fill with random values.
+    RANDOM_INT ('random_int'): Fill with random integers.
+
+    Examples
+    --------
+    data.fill(FillTypes.random)
+    data.fill("random")
+    """
+
+    RANDOM = "random"
+    RANDOM_INT = "random_int"
+
+class UnitsAngles(_LabelsBase):
```

Comment:
My preference would be AngleUnits

everything else has the thing it is first, e.g. Acquisition / Fill etc. so I think this would follow the same pattern

Reply:
we agreed there would be more units in future so keep Units out front